app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -269,18 +268,16 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun initDraftAndViewModel() {
-
-        newMessageViewModel.initDraftAndViewModel(
-            intent = requireActivity().intent,
-            navArgs = newMessageActivityArgs,
-        ).observe(viewLifecycleOwner) { isSuccess ->
-            if (isSuccess) {
-                showKeyboardInCorrectView()
-            } else {
-                requireActivity().apply {
-                    showToast(R.string.failToOpenDraft)
-                    finish()
+    private fun initDraftAndViewModel() = with(newMessageViewModel) {
+        if (initResult.value == null) {
+            initDraftAndViewModel(intent = requireActivity().intent).observe(viewLifecycleOwner) { isSuccess ->
```

Comment:
Maybe another method name? We're in `initDraftAndViewModel` and we call another `initDraftAndViewModel`.